modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -219,59 +192,43 @@ private String getVersion(Extension extension)
         return versionResolver.resolveVersion(extension);
     }
 
-    private void declareConfigurations(ExtensionDeclarer declaration, Class<?> extensionType)
+    private void declareConfigurations(ExtensionDeclarer declaration, ExtensionType<?> extensionType)
     {
-        Class<?>[] configurationClasses = getConfigurationClasses(extensionType);
-        if (ArrayUtils.isEmpty(configurationClasses))
+        List<TypeBasedComponent<?>> configurationClasses = extensionType.getConfigurations();
+        if (configurationClasses.isEmpty())
         {
             declareConfiguration(declaration, extensionType, extensionType);
         }
         else
         {
-            for (Class<?> configurationClass : configurationClasses)
+            for (TypeBasedComponent<?> configurationClass : configurationClasses)
             {
                 declareConfiguration(declaration, extensionType, configurationClass);
             }
         }
     }
 
-    private Class<?>[] getConfigurationClasses(Class<?> extensionType)
-    {
-        Configurations configs = extensionType.getAnnotation(Configurations.class);
-        return configs == null ? ArrayUtils.EMPTY_CLASS_ARRAY : configs.value();
-    }
-
-    private void declareMessageSources(HasSourceDeclarer declarer, Class<?> extensionType)
-    {
-        Sources sources = extensionType.getAnnotation(Sources.class);
-        if (sources != null)
-        {
-            for (Class<? extends Source> declaringClass : sources.value())
-            {
-                declareMessageSource(declarer, declaringClass);
-            }
-        }
-    }
-
-    private void declareConfiguration(ExtensionDeclarer declaration, Class<?> extensionType, Class<?> configurationType)
+    private void declareConfiguration(ExtensionDeclarer declaration, TypeBasedComponent<?> extensionType, TypeBasedComponent<?> configurationType)
     {
-        checkConfigurationIsNotAnOperation(configurationType);
+        checkConfigurationIsNotAnOperation(configurationType.getDeclaredClass());
```

Comment:
will you be able to obtain an actual class when you use the AST version? Won't the class be non existing yet?